src/GUI/RevEng.Core/Tables/CSharpModelGeneratorExtend.cs

```diff
@@ -0,0 +1,115 @@
+        ///     the same compatibility standards as public APIs. The base CSharpModelGenerator may be changed or removed without 
+        ///     notice in any release. Updating Entity Framework Core version in project may result application failures.
+        /// </summary>
+        public override ScaffoldedModel GenerateModel(
+            [NotNull] IModel model,
+            [NotNull] ModelCodeGenerationOptions options)
+        {
+            if (options.ContextName == null)
+            {
+                throw new ArgumentException(
+                    CoreStrings.ArgumentPropertyNull(nameof(options.ContextName), nameof(options)), nameof(options));
+            }
+
+            if (options.ConnectionString == null)
+            {
+                throw new ArgumentException(
+                    CoreStrings.ArgumentPropertyNull(nameof(options.ConnectionString), nameof(options)), nameof(options));
+            }
+
+            // TODO: Honor options.Nullable (issue #15520)
+            var generatedCode = CSharpDbContextGenerator.WriteCode(
+                model,
+                options.ContextName,
+                options.ConnectionString,
+                options.ContextNamespace,
+                options.ModelNamespace,
+                options.UseDataAnnotations,
+                options.SuppressConnectionStringWarning
+#if CORE50 || CORE60
+                ,options.SuppressOnConfiguring
+#endif
+                );
+            
+            // output DbContext .cs file
+            var dbContextFileName = options.ContextName + FileExtension;
+            var resultingFiles = new ScaffoldedModel
+            {
+                ContextFile = new ScaffoldedFile
+                {
+                    Path = options.ContextDir != null
+                        ? Path.Combine(options.ContextDir, dbContextFileName)
+                        : dbContextFileName,
+                    Code = generatedCode
+                }
+            };
+
+            foreach (var entityType in model.GetEntityTypes())
+            {
+                // TODO: Honor options.Nullable (issue #15520)
+                generatedCode = CSharpEntityTypeGenerator.WriteCode(entityType, options.ModelNamespace, options.UseDataAnnotations);
+
+                // output EntityType poco .cs file
+                var entityTypeFileName = entityType.Name + FileExtension;
+                resultingFiles.AdditionalFiles.Add(
+                    new ScaffoldedFile 
+                    {
+                        Path = ReverseEngineerCommandOptions.UseSchemaFolders
+                            ? Path.Combine(entityType.FindAnnotation("Relational:Schema")?.Value.ToString() ??
+                                           entityType.FindAnnotation("Relational:ViewSchema")?.Value.ToString() ??
+                                           "dbo", entityTypeFileName)
```

Comment:
Why this special casing for SQL Server? Seems not needed.

Reply:
This is the only way where you can get information about table/view from annotations. If you remove this part you will not have schema name for tables and views and it will work only for Stored Procedures.

Reply:
There is this (see CommentCSharpEntityTypeGenerator):
```
            var schema = entityType.GetSchema();
            var defaultSchema = entityType.Model.GetDefaultSchema();
```
But what I meant that there is SQL Server specific schema name ("dbo"), which should not be there. There are some database systems that do not even have schemas, and that should just fall back to entitypeFileName.

I think this could be simplified to:
```
var schema = entityType.GetSchema();
Path = ReverseEngineerCommandOptions.UseSchemaFolders && !string.IsNullOrEmpty(schema) 
? Path.Combine(schema, entityTypeFileName)
: entitypeFileName

```

Reply:
I wonder if this could simply be done here: https://github.com/ErikEJ/EFCorePowerTools/blob/master/src/GUI/RevEng.Core/ReverseEngineerScaffolder.cs#L220 (so no need to override EF Core class)

Reply:
Improved this part in the latest commit `entityType.GetSchema()` returns dbo value for the views after EF Core 5.0 even you are using some specific schema. For that purpose, you have to call `entityType.GetViewSchema()` in the versions EF Core 5.0 and 6.0. You can see changes in the latest commit.

Reply:
Regarding your message to try to not override the EF Core class, I would be happy if we can do that. However, in the line [https://github.com/ErikEJ/EFCorePowerTools/blob/master/src/GUI/RevEng.Core/ReverseEngineerScaffolder.cs#L220](https://github.com/ErikEJ/EFCorePowerTools/blob/master/src/GUI/RevEng.Core/ReverseEngineerScaffolder.cs#L220) we should iterate again through the model and split the directory from the file in the path to put schema information, which can be a little bit insecure.

As I said it would be better to not override the EF Core class because as you can see they still have some TO DO comments in the core one.

Reply:
Yes, but stored procedure results are not part of this use case.

Reply:
I will try to remove the override when this has been merged (or you could do it in a follow up PR?)